### PR TITLE
Make workspace folder modification API stable

### LINKED ISSUE #44046: Make multi root modification APIs stable
Currently they are proposed API. 

@jrieken any objections?

### PR DESCRIPTION
**Pull request by bpasero**
fixes #44046

### CODE REVIEW

File: src/vs/vscode.d.ts [hunk 1]
```diff
@@ -5434,6 +5434,49 @@ declare module 'vscode' {
...
+		 *
+		 * @param start the zero-based location in the list of currently opened [workspace folders](#WorkspaceFolder)
+		 * from which to start deleting workspace folders.
+		 * @param deleteCount the optional number of workspace folders to remove.
+		 * @param workspaceFoldersToAdd the optional variable set of workspace folders to add in place of the deleted ones.
+		 * Each workspace is identified with a mandatory URI and an optional name.
+		 * @return true if the operation was successfully started and false otherwise if arguments were used that would result
+		 * in invalid workspace folder state (e.g. 2 folders with the same URI).
+		 */
+		export function updateWorkspaceFolders(start: number, deleteCount: number, ...workspaceFoldersToAdd: { uri: Uri, name?: string }[]): boolean;
```
Comment (jrieken) [inline src/vs/vscode.d.ts]: We should have `deleteCount: number | undefined | null` to be compliant with strict-null-checks

Comment (jrieken) [reply src/vs/vscode.d.ts]: We could also use the same Array.splice-semantics on `start` which is that start = -1 is adding to the end.

Comment (jrieken) [reply src/vs/vscode.d.ts]: An last, we could have another overload without `deleteCount` (this or have it optional) like so: 

`export function updateWorkspaceFolders(start: number, deleteCount: number, ...workspaceFoldersToAdd: { uri: Uri, name?: string }[]): boolean;`

`export function updateWorkspaceFolders(start: number, ...workspaceFoldersToAdd: { uri: Uri, name?: string }[]): boolean;`

Comment (bpasero) [reply src/vs/vscode.d.ts]: @jrieken 

> We should have deleteCount: number | undefined | null to be compliant with strict-null-checks
I added that.

As for the rest I have opened https://github.com/Microsoft/vscode/issues/44570 for the future. There seems to be more `splice` semantics that we could look into also for the `deleteCount` being negative.

**Pull request merged**